### PR TITLE
Fix email translations lost and overwritten by parent

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✅ 
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This is just minor addition to use flag updatingTranslationChildren in plugins. It help us prevent some entity save content during process entities via  `EmailEvents::EMAIL_POST_SAVE`

#### Steps to test this PR:

1. Load up [this PR](https://mautibox.com)
2. Just code review. Nothing more

### CODE REVIEW

**Review by mohit-rocks — DISMISSED** [review]

**Comment by mohit-rocks** [comment]
Closing and reopening it to trigger the test bot again.

File: app/bundles/EmailBundle/Tests/Model/EmailModelTest.php [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 namespace Mautic\EmailBundle\Tests\Model;
 
+define('SORTARRIVAL', 1);
```
Comment (escopecz) [inline app/bundles/EmailBundle/Tests/Model/EmailModelTest.php]: Why is this needed here?

**Review by escopecz — DISMISSED** [review]
The code looks good. I'm not sure about one row. Please let me know why it is there.

File: app/bundles/EmailBundle/Tests/Model/EmailModelTest.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/EmailBundle/Tests/Model/EmailModelTest.php]: It return error undefined while I tested it on local enviroment. I forgot removed it. It's removed it

**Review by alanhartless — DISMISSED** [review]
The only thing I'd call out is that the new testIsUpdatingTranslationChildren is only testing the negative (assertFalse) but not the positive (assertTrue). 

I know that the model only has access after the code is ran. To better cover the new method, I think you could instantiate `new EventDispatcher()` in a second test instead of mocking it. Mock/create a couple email entities (you'll have to be able to programmatically determine that one of them is a translation likely through getTranslation()), create and register a listener class to EmailEvents::EMAIL_POST_SAVE that checks if the email is one of the variants and if so, assert that EmailModel::isUpdatingTranslationChildren() returns true.

**Review by fedys — DISMISSED** [review]
There are no issues apart from the Alan's last comment. I added the assertions in 4a1360a35fb86e92519556f26345a15ac7410bdc.

**Review by dennisameling — APPROVED** [review]
Approving and merging since the feedback has been addressed now 👍🏼

**Comment by garzettone** [comment]
Hi, I just updated to 4.02 release and the problem of translation override is still there.

**Comment by RCheesley** [comment]
> Hi, I just updated to 4.02 release and the problem of translation override is still there.

Please can you raise a new issue? Thanks!

**Pull request merged**